src/goto-symex/solver_hardness.h

```diff
@@ -160,4 +161,26 @@ struct hash<solver_hardnesst::hardness_ssa_keyt>
 };
 } // namespace std
 
+static inline void with_solver_hardness(
+  decision_proceduret &maybe_hardness_collector,
+  std::function<void(solver_hardnesst &hardness)> handler)
+{
+  // FIXME I am wondering if there is a way to do this that is a bit less
+  // dynamically typed.
+  if(
+    auto prop_conv_solver =
+      dynamic_cast<prop_conv_solvert *>(&maybe_hardness_collector))
+  {
+    if(auto hardness_collector = prop_conv_solver->get_hardness_collector())
+    {
+      if(hardness_collector->solver_hardness)
+      {
+        auto &solver_hardness = static_cast<solver_hardnesst &>(
+          *(hardness_collector->solver_hardness));
+        handler(solver_hardness);
+      }
+    }
+  }
```

Comment:
This surely is better already, but don't we just need a Boolean function like `virtual bool set_hardness_collector(solver_hardnesst *) { return false; }` in `prop_conv_solvert` that will return `true` when the hardness collector was accepted?

Reply:
To make the above go away, you'd need a get_solver_hardness_collector, or the like, in `decision_proceduret`. I can't get myself to do that.